tests/bot/cogs/moderation/test_utils.py

```diff
@@ -0,0 +1,477 @@
+
+                result = await post_user(*args)
+
+                if error:
+                    self.assertIsNone(result)
+                else:
+                    self.assertEqual(result, expected)
+
+                if not error:
+                    self.bot.api_client.post.assert_awaited_once_with("bot/users", json=payload)
+
+                self.bot.api_client.post.reset_mock(side_effect=True)
+
+    async def test_send_private_embed(self):
+        """Test does `send_private_embed` return correct value."""
+        test_cases = [
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": True,
+                "raised_exception": None
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": HTTPException(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": Forbidden(AsyncMock(), AsyncMock())
+            },
+            {
+                "args": (self.user, Embed(title="Test", description="Test val")),
+                "expected_output": False,
+                "raised_exception": NotFound(AsyncMock(), AsyncMock())
+            }
+        ]
+
+        for case in test_cases:
+            args = case["args"]
+            expected = case["expected_output"]
+            raised: Union[Forbidden, HTTPException, NotFound, None] = case["raised_exception"]
+
+            with self.subTest(args=args, expected=expected, raised=raised):
+                if raised:
+                    self.user.send.side_effect = raised
+
+                result = await send_private_embed(*args)
+
+                self.assertEqual(result, expected)
+                if expected:
+                    args[0].send.assert_awaited_once_with(embed=args[1])
+
+                self.user.send.reset_mock(side_effect=True)
+
+    @patch("bot.cogs.moderation.utils.post_user")
+    async def test_post_infraction(self, post_user_mock):
+        """Test does `post_infraction` return correct value."""
+        now = datetime.now()
+        test_cases = [
+            {
+                "args": (self.ctx, self.member, "ban", "Test Ban"),
+                "expected_output": [
+                    {
+                        "id": 1,
+                        "inserted_at": "2018-11-22T07:24:06.132307Z",
+                        "expires_at": "5018-11-20T15:52:00Z",
+                        "active": True,
+                        "user": 1234,
+                        "actor": 1234,
+                        "type": "ban",
+                        "reason": "Test Ban",
+                        "hidden": False
+                    }
+                ],
```

Comment:
As I pointed out elsewhere, the data doesn't need to be this accurate. Using "dummy value" would achieve the same results since `post_infraction` just returns whatever `api_client.post` returns.